WcaOnRails/app/helpers/application_helper.rb

```diff
@@ -48,14 +48,21 @@ def anchorable(pretty_text, id=nil)
 
   WCA_EXCERPT_RADIUS = 50
 
-  def wca_excerpt(html, phrase)
+  def wca_excerpt(html, phrases)
     text = ActiveSupport::Inflector.transliterate(strip_tags(html)) # TODO https://github.com/cubing/worldcubeassociation.org/issues/238
-    excerpted = excerpt(text, phrase, radius: WCA_EXCERPT_RADIUS)
-    # If nothing matches the given phrase, just return the beginning.
-    if excerpted.blank?
-      excerpted = truncate(text, length: WCA_EXCERPT_RADIUS)
-    end
-    wca_highlight(excerpted, phrase)
+    # Compute the first and last index where query parts appear and use the whole text between them for excerpt.
+    first = phrases.map { |phrase| text.index(phrase) }.compact.min
```

Comment:
We remove all `nil` values by calling `compact` to avoid number and nil comparision error.
`[].min` returns `nil` and in our case indicates that nothing matches the given text.